pkg/api/helpers.go

```diff
@@ -41,3 +44,31 @@ func GetFieldLabelConversionFunc(supportedLabels map[string]string, overrideLabe
 func GetResourceKey(obj metav1.ObjectMeta) string {
 	return fmt.Sprintf("%s/%s", obj.Namespace, obj.Name)
 }
+
+// OriginObjects returns the origin types.
+func OriginObjects(objs []runtime.Object) []runtime.Object {
```

Comment:
really not a fan of these functions in general, and definitely not in places non-test code can call them. can we put them under a fake/test package if they're required?